bin/desi_qso_mgii_afterburner

```diff
@@ -189,14 +190,20 @@ def save_dataframe_to_fits(dataframe, filename, DESI_TARGET, clobber=True):
     data['VAR_SIGMA'] = dataframe['VAR_SIGMA']
     data['VAR_B'] = dataframe['VAR_B']
 
-    fits = fitsio.FITS(filename, 'rw', clobber=clobber)
+    # Save file in temporary file to track when timeout error appears during the writing
+    tmpfile = get_tempfilename(filename)
+    fits = fitsio.FITS(tmpfile, 'rw', clobber=clobber)
     fits.write(data, extname='MGII')
     if clobber:
         log.info(f'OVERWRITE the file : {filename}')
     else:
         log.info(f'EXPAND the file : {filename}')
     fits.close()
```

Comment:
Since tmpfile will always be unique, this code will no longer append to a pre-existing filename and will effectively always clobber the pre-existing file with the `os.rename(tmpfile, filename)`.

What is the intended behavior here?  Do you ever really want to append another identically named "MGII" extension to a pre-existing file?  Or maybe load the pre-existing file, append to the table, and then write a new single "MGII" extension?  Or do you just want the code to stop if the filename already exists and doesn't have --clobber?

To get the original behavior (writing another extension), you'll need to do somthing like:
```
if os.path.exists(filename) and not clobber:
    os.rename(filename, tmpfile)

fits = fitios.FITS(tmpfile, ...)
...
fits.close()
os.rename(tmpfile, filename)
```

Reply:
Thank for this remark !  This is now useless (put here for testing purpose), I remove the clobber option.